knox/knox.sh

```diff
@@ -128,8 +160,31 @@ function initialize_crontab() {
 function install() {
   local master_name
   master_name=$(/usr/share/google/get_metadata_value attributes/dataproc-master)
-  if [[ "${master_name}" == $(hostname) ]]; then
+  if [[ "${DATAPROC_IMAGE_VERSION}" == "2.2" ]]; then
+      if [[ "${master_name}" == $(hostname -A | tr -d '[:space:]' | awk -F "." '{print $1}') ]]; then
+         [[ -z "${KNOX_GW_CONFIG_GCS}" ]] && err "Metadata knox-gw-config is not provided. knox-gw-config stores the configuration files for knox."
+          
+          retry_command "apt-get update"
+          install_dependencies
+          update
+          enable_demo_ldap_for_dev_testing
+          get_config_parameters
+          replace_the_master_key
+          generate_or_replace_certificate
+          initialize_crontab
+
+         [[ -f "/lib/systemd/system/knoxldapdemo.service" ]] && systemctl restart knoxldapdemo
+         systemctl restart knox
+      fi
+  elif [[ "${master_name}" == $(hostname) ]]; then
     [[ -z "${KNOX_GW_CONFIG_GCS}" ]] && err "Metadata knox-gw-config is not provided. knox-gw-config stores the configuration files for knox."
+      if [ "${DATAPROC_IMAGE_VERSION}" = "2.0" ]; then
+        echo "Dataproc version is 2.0"
+        SUB_MINOR_VERSION=$(get_dataproc_sub_minor_version)
+        if [ "${SUB_MINOR_VERSION}" -le 102 ]; then
+           comment_repo_line
+        fi
+      fi
```

Comment:
Why couldn't that 2.2 block above instead nest here as an `elif [[ "${DATAPROC_IMAGE_VERSION}" == "2.2" ]]; then` ?  I think it would be more appropriate to put it here and leave the initial condition as it was.  If you want to change the `if [[ "${master_name}" == $(hostname) ]]; then` to `if [[ "${master_name}" == $(hostname -s) ]]; then` that would be a reasonable change, too.